task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -296,24 +302,48 @@ def xcom_pull(
             task_ids = [self.task_id]
         elif isinstance(task_ids, str):
             task_ids = [task_ids]
+
+        map_indexes_iterable: Iterable[int | None] = []
+        # If map_indexes is not provided, default to use the map_index of the calling task
         if isinstance(map_indexes, ArgNotSet):
-            map_indexes = self.map_index
+            map_indexes_iterable = [self.map_index]
+        elif isinstance(map_indexes, int) or map_indexes is None:
+            map_indexes_iterable = [map_indexes]
         elif isinstance(map_indexes, Iterable):
-            # TODO: Handle multiple map_indexes or remove support
-            raise NotImplementedError("Multiple map_indexes are not supported yet")
+            map_indexes_iterable = map_indexes
+        else:
+            raise TypeError(
+                "map_indexes can be omitted or must be an int, an iterable of ints, "
+                f"or None, got {type(map_indexes)}"
+            )
 
         log = structlog.get_logger(logger_name="task")
 
         xcoms = []
-        for t in task_ids:
+        # TODO: Execution API only allows working with a single map_index at a time
```

Comment:
```suggestion
        # TODO: AIP 72 Execution API only allows working with a single map_index at a time
```

We are adding the AIP 72 as we have done it for multiple places in the repo, easier to grep at a later point